### PR TITLE
Do not issue update command if nothing has changed in user values

### PR DESCRIPTION
**Pull request by MorrisJobke**
Same as #14566 but rebased on master so that we can debug the failing CI.

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Let me debug this later today

**Comment by MorrisJobke** [comment]
Partially added via #15052 to properly debug the CI failure in smaller steps.

File: tests/lib/Util/User/Dummy.php [hunk 1]
```diff
@@ -164,6 +164,7 @@ public function countUsers() {
 
 	public function setDisplayName($uid, $displayName) {
 		$this->displayNames[$uid] = $displayName;
+		return true;
```
Comment (MorrisJobke) [inline tests/lib/Util/User/Dummy.php]: I found the cause of the failure - no idea why it worked before, but it was also failing locally and this fixed it locally for me as well 🙌

**Review by rullzer — APPROVED** [review]
But of course :confused:

**Review by ChristophWurst — APPROVED** [review]

File: tests/lib/Util/User/Dummy.php (hunk 1 above)
Comment (MorrisJobke) [reply tests/lib/Util/User/Dummy.php]: https://github.com/nextcloud/server/pull/14566#issuecomment-479118224

File: lib/private/User/User.php [hunk 2]
```diff
@@ -138,11 +138,12 @@ public function getDisplayName() {
 	 */
 	public function setDisplayName($displayName) {
 		$displayName = trim($displayName);
-		if ($this->backend->implementsActions(Backend::SET_DISPLAYNAME) && !empty($displayName)) {
+		$oldDisplayName = $this->getDisplayName();
+		if ($this->backend->implementsActions(Backend::SET_DISPLAYNAME) && !empty($displayName) && $displayName !== $oldDisplayName) {
```
Comment (kesselb) [inline lib/private/User/User.php]: Make sense. Just a small nitpick: `!empty($displayName) && $displayName !== $oldDisplayName` is cheaper than `implementsActions`.

Comment (MorrisJobke) [reply lib/private/User/User.php]: Something for later I would say.

**Pull request merged**